linux/x86-64/python3/acrcloud/async_recognizer.py

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env python
+#-*- coding:utf-8 -*-
+
+'''
+    @author qinxue.pan E-mail: xue@acrcloud.com
+    @version 1.0.0
+    @create 2015.10.01
+'''
+
+import os
+import sys
+import hmac
+import time
+import json
+import base64
+import hashlib
+import aiohttp
+import async_timeout
+import datetime
+
+import acrcloud_extr_tool
+
+'''
+Copyright 2015 ACRCloud Recognizer v1.0.0
+
+This module can recognize ACRCloud by most of audio/video file. 
+        Audio: mp3, wav, m4a, flac, aac, amr, ape, ogg ...
+        Video: mp4, mkv, wmv, flv, ts, avi ...
+
+Example:
+    config = {
+        'host':'ap-southeast-1.api.acrcloud.com',
+        'access_key':'XXXXXXXX',
+        'access_secret':'XXXXXXXX',
+        'timeout':5
+    }
+    re = ACRCloudRecognizer(config)
+
+    #recognize by file path, and skip 180 seconds from from the beginning of "aa.mp3".
+    print re.recognize_by_file('aa.mp3', 180)
+
+    buf = open('aa.mp3', 'rb').read()
+    #recognize by file_audio_buffer that read from file path, and skip 180 seconds from from the beginning of "aa.mp3".
+    print re.recognize_by_filebuffer(buf, 180)
+
+    #aa.wav is (RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 8000 Hz)
+    buf = open('aa.wav', 'rb').read()
+    buft = buf[1024000:192000+1024001]
+    recognize by audio_buffer(RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 8000 Hz)
+    print re.recognize(buft)
+'''
+class ACRCloudRecognizeType:
+    ACR_OPT_REC_AUDIO = 0  # audio fingerprint
+    ACR_OPT_REC_HUMMING = 1 # humming fingerprint
+    ACR_OPT_REC_BOTH = 2 # audio and humming fingerprint
+
+class ACRCloudRecognizer:
+    def __init__(self, config, loop):
```

Comment:
Best practice here is to do `loop=None` and `loop = loop if loop else asyncio.get_event_loop()`.  You might even make loop a keyword-only argument.